elastica/joint.py

```diff
@@ -954,6 +955,18 @@ def __init__(self, k, nu, velocity_damping_coefficient=0, friction_coefficient=0
         super().__init__(k, nu)
         self.velocity_damping_coefficient = velocity_damping_coefficient
         self.friction_coefficient = friction_coefficient
+        log = logging.getLogger()
+        log.warning(
+            # Remove warning and add error if ExternalContact is used in v0.3.3
+            # Remove the option to use ExternalContact, beyond v0.4.0
+            "The option to use the ExternalContact joint for the rod-rod and rod-cylinder "
+            "contact is now deprecated. Instead, for rod-rod contact or rod-cylinder contact, "
+            "use RodRodContact or RodCylinderContact from the add-on Contact mixin class. "
+            "For reference see the classes elastica.contact_forces.RodRodContact() and elastica.contact_forces.RodCylinderContact(),"
+            " and for usage check examples/RigidbodyCases/RodRigidBodyContact/rod_cylinder_contact.py and examples/RodContactCase/RodRodContact/rod_rod_contact_parallel_validation.py."
+            " The option to use the ExternalContact joint for the rod-rod and rod-cylinder"
+            " will be removed in the future (v0.3.3)."
+        )
```

Comment:
I think all this message will be prompted in one line. Could you please check and divide into couple of lines for readability.

Reply:
<img width="1173" alt="Screenshot 2023-08-20 at 2 26 20 AM" src="https://github.com/GazzolaLab/PyElastica/assets/47090295/aab3f28d-cb76-4838-b41b-36dd60848bac">